src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -123,6 +151,81 @@ export const Configurator = ({
           >
             <EuiFieldText disabled value={dataset.title} />
           </EuiFormRow>
+          {indexedViewsService && (
+            <>
+              <EuiSpacer />
+              <EuiSwitch
+                compressed
+                checked={selectIndexedView}
+                label={
+                  <EuiFormLabel>
+                    {i18n.translate(
+                      'data.explorer.datasetSelector.advancedSelector.configurator.showAvailableIndexedViewsLabel',
+                      {
+                        defaultMessage: 'Query indexed view',
+                      }
+                    )}
+                  </EuiFormLabel>
+                }
+                onChange={(e) => setSelectIndexedView(e.target.checked)}
+              />
+              <EuiSpacer size="m" />
+              {selectIndexedView && (
+                <EuiFormRow
+                  label={i18n.translate(
+                    'data.explorer.datasetSelector.advancedSelector.configurator.indexedViewLabel',
+                    {
+                      defaultMessage: 'Available indexed views',
+                    }
+                  )}
+                  helpText={i18n.translate(
+                    'data.explorer.datasetSelector.advancedSelector.configurator.indexedViewHelpText',
+                    {
+                      defaultMessage: 'Select an indexed view to speed up your query.',
+                    }
+                  )}
+                >
+                  <EuiSelect
+                    isLoading={isLoadingIndexedViews}
+                    options={indexedViews.map(({ name }) => ({
+                      text: name,
+                      value: name,
+                    }))}
+                    value={selectedIndexedView}
+                    onChange={async (e) => {
+                      const value = e.target.value;
+                      setSelectedIndexedView(value);
+                      let connectedDataSource;
+                      if (dataset.dataSource?.id) {
```

Comment:
isn't dataSource have everything you need? 

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/common/datasets/types.ts#L224

logically if the type is not an INDEX_PATTERN then it seems like we just want to replace the dataset.dataSource.type = DATA_SOURCE

vs making this unique to the saved object if im understanding this correctly